test/test_download.py

```diff
@@ -254,12 +276,49 @@ def try_rm_tcs_files(tcs=None):
         tname = 'test_%s_%d' % (test_case['name'], i)
         i += 1
     test_method = generator(test_case, tname)
-    test_method.__name__ = str(tname)
-    ie_list = test_case.get('add_ie')
-    test_method.add_ie = ie_list and ','.join(ie_list)
-    setattr(TestDownload, test_method.__name__, test_method)
-    del test_method
-
+    ie_list = ','.join(test_case.get('add_ie', []))
+    TestDownload.addTest(test_method, tname, ie_list)
+
+# Py2 compat (should be in compat.py?)
```

Comment:
These are already in compat as `compat_filter` and `compat_map` [here](https://github.com/ytdl-org/youtube-dl/blob/master/youtube_dl/compat.py#L2970).

Reply:
Yes, I subsequently needed them in some other PR (after I received commit privs), so they were added there. Here they should be imported.